### PR TITLE
Couple with ELPA

### PR DESCRIPTION
**Pull request by jeanlucf22**

### CODE REVIEW

**Review by suemni — APPROVED** [review]

**Comment by jeanlucf22** [comment]
any comments? ready to merge...

**Comment by nicolasbock** [comment]
I might be missing something, but I can't figure out how one would call the ELPA version of `bml_diagonalize_distributed2d`.  Are you planning on enabling the new functions in a follow-up PR?

File: src/C-interface/distributed2d/bml_diagonalize_distributed2d_typed.c [hunk 1]
```diff
@@ -288,11 +298,195 @@ void TYPED_FUNC(
...
+#endif
+
+void TYPED_FUNC(
+    bml_diagonalize_distributed2d) (
+    bml_matrix_distributed2d_t * A,
+    void *eigenvalues,
+    bml_matrix_distributed2d_t * eigenvectors)
+{
+#ifdef BML_USE_ELPA
+    TYPED_FUNC(bml_diagonalize_distributed2d_elpa) (A, eigenvalues,
```
Comment (jeanlucf22) [inline src/C-interface/distributed2d/bml_diagonalize_distributed2d_typed.c:479]: Function implementing interface to ELPA is called here

Comment (nicolasbock) [reply src/C-interface/distributed2d/bml_diagonalize_distributed2d_typed.c:479]: Oh right. Thanks for pointing me in the right direction 😄

**Review by nicolasbock — APPROVED** [review]

**Pull request merged**